File: benchmarks/runtime/mixed_workload.lua
local n = 200000
local i = 0
local score = 0
local mode = "a"
while i < n do
  local m = { [i] = i + 1, [i + 10] = i + 2 }
  local v = m[i] or 0
  score = score + v
  if (i & 7) == 0 and mode == "a" then
    score = score + 1
  end
  i = i + 1
end
if score <= 0 then os.exit(1) end
print(score)

File: benchmarks/runtime/mixed_workload.js
let n = 200000;
let i = 0;
let score = 0;
const mode = "a";
while (i < n) {
  const m = new Map([[i, i + 1], [i + 10, i + 2]]);
  const v = m.get(i) || 0;
  score += v;
  if ((i & 7) === 0 && mode === "a") {
    score += 1;
  }
  i += 1;
}
if (score <= 0) process.exit(1);
console.log(score);
